frontend/packages/console-shared/src/components/drawer/Drawer.tsx

```diff
@@ -0,0 +1,128 @@
+import * as React from 'react';
+import { CSSTransition } from 'react-transition-group';
+import { DraggableData, DraggableCore } from 'react-draggable';
+import './Drawer.scss';
+
+type DrawerProps = {
+  /**
+   * Height of the drawer,
+   * should be set when used as controlled component with onChange callback
+   */
+  height?: number;
+  /**
+   * Default Value: 300
+   * default height of the drawer component,
+   * should be set when used as Uncontrolled component
+   */
+  defaultHeight?: number;
+  /**
+   * Defines the minimized state of drawer
+   * false: Height is minimum height (minimized state)
+   * true: Height is greater than minimum height
+   */
+  open?: boolean;
+  /**
+   * Maximum height drawer can be resized to.
+   */
+  maxHeight?: number | string;
+  /**
+   * Set whether the drawer is resizable or not.
+   */
+  resizable?: boolean;
+  /**
+   * Content for the Header of drawer
+   */
+  header: React.ReactNode;
+  /**
+   * This callback is invoked while resizing the drawer.
+   * @param open boolean: false when the drawer reached minimum height (minimized state)
+   * @param height number: Height of the drawer while resizing
+   */
+  onChange?: (open: boolean, height: number) => void;
+};
+
+const useSize = <T extends HTMLElement>(): [number | undefined, (element: T) => void] => {
+  const [height, setHeight] = React.useState<number>(0);
+  const ref = React.useRef<number>();
+
+  const callback = React.useCallback((element: T): void => {
+    if (element) {
+      const bb = element.getBoundingClientRect();
+      if (!ref.current) setHeight(bb.height);
+    }
+  }, []);
+  return [height, callback];
+};
+
+const Drawer: React.FC<DrawerProps> = ({
+  children,
+  defaultHeight = 300,
+  height,
+  maxHeight = '100%',
+  open = true,
+  resizable = false,
+  header,
+  onChange,
+}) => {
+  const [heightState, setHeightState] = React.useState(defaultHeight);
+  const lastObservedHeightRef = React.useRef<number>(0);
+  const [minHeight, ref] = useSize<HTMLDivElement>();
+  const resizeHeight = height ?? heightState;
+  const minimumHeight = minHeight ?? 0;
+  const handleDrag = (e: MouseEvent, data: DraggableData) => {
+    const newHeight = resizeHeight - data.y;
+    if (onChange) {
+      if (newHeight > minimumHeight) {
+        onChange(true, newHeight);
+      } else {
+        onChange(false, minimumHeight);
```

Comment:
When closed, set height to last observed height. That way you don't need to do so on drag stop.

Reply:
same as https://github.com/openshift/console/pull/4724#discussion_r392807610